wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/FromGeneric.hs

```diff
@@ -178,6 +184,34 @@ feeOptions CoinSelectionOptions{..} = FeeOptions{
   Building transactions
 -------------------------------------------------------------------------------}
 
+-- | Our notion of @unsigned transaction@. Unfortunately we cannot reuse
+-- directly the 'Tx' from @Core@ as that discards the information about
+-- "ownership" of inputs, which is instead required when dealing with the
+-- Core Txp.Util API.
+data UnsignedTx = UnsignedTx {
+    utxOwnedInputs :: NonEmpty (Core.TxIn, Core.TxOutAux)
+  , utxOutputs     :: NonEmpty Core.TxOutAux
+  , utxChange      :: [Core.Coin]
+  , utxAttributes  :: Core.TxAttributes
+}
+
+-- | Creates a "standard" unsigned transaction.
+mkStdUnsignedTx :: Monad m
+             => (forall a. NonEmpty a -> m (NonEmpty a))
+             -- ^ Shuffle function
+             -> NonEmpty (Core.TxIn, Core.TxOutAux)
+             -- ^ Selected inputs
+             -> NonEmpty Core.TxOutAux
+             -- ^ Selected outputs
+             -> [Core.Coin]
+             -- ^ Change coins
+             -> m UnsignedTx
+mkStdUnsignedTx shuffle inps outs change = do
+    allOuts <- shuffle outs
+    let Core.UnsafeTx{..} = CTxp.makeUnsignedAbstractTx (fmap repack inps) allOuts
```

Comment:
I feel that this call to `makeUnsignedAbstractTx` here with a subset of the final outputs is somewhat misleading. We don't know exactly what that attribute calculation will depend on, and calling this function with a subset of all the input data feels somewhat dangerous. Let's just remove `utxAttributes` from `UnsignedTx`. 

It's not entirely clear to me what the right input is for the external ledger device; it depends on where we use hardened and non-hardened derivation. Let's not try to anticipate.